services/web-server/src/login/AzureSessionStore.js

```diff
@@ -0,0 +1,98 @@
+const assert = require('assert');
+const taskcluster = require('taskcluster-client');
+const tryCatch = require('../utils/tryCatch');
+const { NOOP } = require('../utils/constants');
+
+module.exports = function ({ session, SessionStorage, options = {} }) {
+  const { Store }= session;
+
+  assert(session, 'An express-session object is required');
+  assert(SessionStorage, 'A session storage is required');
+
+  const {
+    // Session timeout in a format that `fromNow` (taskcluster-client) understands. Defaults to 1 day
+    sessionTimeout = '1 day',
+  } = options;
+
+  return class AzureSessionStore extends Store {
+    /*
+     * required
+     *
+     * This is used to destroy/delete a session from the store given a session ID (sessionId).
+     * The callback should be called as callback(error) once the session is destroyed.
+     */
+    async destroy(sessionId, callback = NOOP) {
+      await SessionStorage.remove({ sessionId }, true);
```

Comment:
If this fails, it will be an UnhandledPromiseRejection and the callback will never be called.  I feel like we've seen this issue in a few places (oauth2.js comes to mind).  Maybe there's a general solution for de-promisifying a function like this?

Reply:
Good observation. The immediate fix that comes to mind is to wrap the method with a try catch method.